zallet/src/components/json_rpc/methods/list_unspent.rs

```diff
@@ -204,6 +213,64 @@ pub(crate) fn call(
 
         let is_watch_only = !matches!(account.purpose(), AccountPurpose::Spending { .. });
 
+        let utxos = wallet
+            .get_transparent_receivers(account_id, true)
+            .map_err(|e| {
+                RpcError::owned(
+                    LegacyCode::Database.into(),
+                    "WalletDb::get_transparent_receivers failed",
+                    Some(format!("{e}")),
+                )
+            })?
+            .iter()
+            .try_fold(vec![], |mut acc, (addr, _)| {
+                let mut outputs = wallet
+                    .get_spendable_transparent_outputs(addr, target_height, confirmations_policy)
+                    .map_err(|e| {
+                        RpcError::owned(
+                            LegacyCode::Database.into(),
+                            "WalletDb::get_spendable_transparent_outputs failed",
+                            Some(format!("{e}")),
+                        )
+                    })?;
+
+                acc.append(&mut outputs);
+                Ok::<_, RpcError>(acc)
+            })?;
+
+        for utxo in utxos {
+            let confirmations = utxo.mined_height().map(|h| target_height - h).unwrap_or(0);
+
+            let wallet_internal = wallet
+                .get_transparent_address_metadata(account_id, utxo.recipient_address())
+                .map_err(|e| {
+                    RpcError::owned(
+                        LegacyCode::Database.into(),
+                        "WalletDb::get_transparent_address_metadata failed",
+                        Some(format!("{e}")),
+                    )
+                })?
+                .is_some_and(|m| m.scope() == TransparentKeyScope::INTERNAL);
```

Comment:
Hmm: what does it mean if we know the account for a transparent address, but we don't have its metadata? That is, should `walletInternal = false` mean either "external" or "we don't know", or should this be considered an error case, or should `walletInternal` be optional (and omitted if we cannot determine it)?